app/views/episodes/_form_audio.html.erb

```diff
@@ -1,10 +1,47 @@
-<h2><%= t(".title") %></h2>
+<div class="prx-uploads col-12 mb-4">
+  <div class="card shadow border-0">
 
-<div class="col-4 mb-4 prx-field-group">
-  <div class="form-floating">
-    <%= form.number_field :segment_count, min: 1, max: 10 %>
-    <%= form.label :segment_count %>
-  </div>
+    <div class="card-header card-header-info d-flex justify-content-between">
+      <h5 class="card-title"><%= t(".title") %></h5>
+    </div>
+
+    <div class="card-body">
+      <%= turbo_frame_tag "episode-form-audio", data: {controller: "audio-upload"} do %>
+        <div class="row">
+
+          <div class="col-12 mb-4 prx-field-group">
+            <div class="form-floating" data-controller="dynamic-form">
+              <%= form.number_field :segment_count, min: 1, max: Episode::MAX_SEGMENT_COUNT, data: {action: "dynamic-form#change"} %>
+              <%= form.label :segment_count %>
+              <a hidden href="<%= request.fullpath %>" data-dynamic-form-target="link"></a>
+            </div>
+
+            <button type="button" class="prx-btn-help" data-bs-toggle="popover" data-bs-content="<%= t(".help.segment_count") %>">
+              <span class="material-icons">help</span>
+            </button>
+          </div>
 
-  <button type="button" class="prx-btn-help" data-bs-toggle="popover" data-bs-content="<%= t(".help.segment_count") %>"><span class="material-icons">help</span></button>
+          <%= form.fields_for :contents, episode.build_contents, include_id: false do |fields| %>
+            <% content = fields.object %>
+
+            <%= turbo_frame_tag "episode-form-audio-#{content.position}" do %>
```

Comment:
Okay, and this partial has all the turbo-magic in it:

1. An outer parent turbo-frame, which the `segment_count` field updates (which adds/removes contents)
2. Many inner content turbo-frames, for each `position` in this segment_count.  Uploading/deleting/replacing the content in that position only impacts that inner turbo-frame.